metpy/calc/tools.py

```diff
@@ -1375,3 +1375,132 @@ def _abbrieviate_direction(ext_dir_str):
+    r"""Performs horizontal area-averaging of a field in latitude/longitude format over a specified radius.
+    Parameters
+    ----------
+    var : (M, N) ndarray
+        Variable to perform area averaging on. Can be 2, 3 or 4 dimensions. If 2D, coordinates must
+        be lat/lon. If using additional dimensions, area-averaging will only be performed on the last
+        2 dimensions, assuming those are latitude and longitude.
+    rad : `pint.Quantity`
+        The radius over which to perform the spatial area-averaging.
+    lon : array-like
+        Array of longitudes defining the grid
+    lat : array-like
+        Array of latitudes defining the grid
+        
+    Returns
+    -------
+    (M, N) ndarray
+        Area-averaged quantity, returned in the same dimensions as passed.
+    
+    Notes
+    -----
+    This function was originally provided by Matthew Janiga and Philippe Papin using a Fortran wrapper for NCL,
+    and converted to python with further efficiency modifications by Tomer Burg, with permission from the original
+    authors.
+    
+    This function assumes that the last 2 dimensions of var are ordered as (....,lat,lon). This can perhaps be modified
+    to handle (...,lon,lat) as well.
+    """
+    
+    #Inner function to perform the area-averaging
+    def calcavg(x,xavg,lon2d,lat2d,nlon,nlat,rad,box,eqrm):
+        nbox = (2*box+1)*(2*box+1)
+
+        #Iterate over latitude and longitude
+        for j in range((box),(nlon-box)):
+            for i in range((box),(nlat-box)):
+
+                lon1d = lon2d[i-box:i+box+1,j-box:j+box+1].reshape((nbox))
+                lat1d = lat2d[i-box:i+box+1,j-box:j+box+1].reshape((nbox))
+                x1d = x[i-box:i+box+1,j-box:j+box+1].reshape((nbox))
+
+                d1d = eqrm * np.sqrt(( (lon2d[i,j]-lon1d)*np.cos( (lat2d[i,j]+lat1d)/2.0 ) )**2.0 + (lat2d[i,j]-lat1d)**2.0)
+                z = x1d[d1d < rad] / len(x1d[d1d < rad])
+                xavg[i,j] = np.sum(z)
+
+        return xavg
+    
+    #convert passed radius to kilometers
+    rad = rad.to('kilometers')
+    
+    #Calculate distance in km of dataset resolution, at the equator, given lat/lon arrays
+    londiff = lon[1]-lon[0]
+    latdiff = lat[1]-lat[0]
+    lat_0 = 0.0 - (latdiff/2.0)
+    lat_1 = 0.0 + (latdiff/2.0)
+    dx,dy = lat_lon_grid_deltas(np.array([lon[0],lon[1]]), np.array([lat_0,lat_1]))
+    dx = dx.to('km')
+    res = int((dx[0].magnitude + dx[1].magnitude)/2.0) * units('km')
+    
+    #---------------------------------------------------------------------
+    #Error checks
+    
+    #Check to make sure latitudes increase
+    reversed_lat = 0
+    if lat[1] < lat[0]:
+        reversed_lat = 1
+        
+        #Reverse latitude array
+        lat = lat[::-1]
+        
+        #Determine which axis of variable array to reverse
+        lat_dim = len(var.shape)-2
+        var = np.flip(var,lat_dim)
+        
+    #Check to ensure input array has 2, 3 or 4 dimensions
```

Comment:
I'd raise an error here - we generally don't have print statements in a library, just errors and log messages.